pkg/controller/egress/controller_test.go

```diff
@@ -136,6 +139,45 @@ type egressController struct {
 func newController(objects, crdObjects []runtime.Object) *egressController {
 	client := fake.NewSimpleClientset(objects...)
 	crdClient := fakeversioned.NewSimpleClientset(crdObjects...)
+	// These reactors are in charge of incrementing Generation for Egress resources, so that
+	// changes done through crdClient are not ignored by the EgressController.
+	egressUpdateReactor, egressPatchReactor := func() (k8stesting.ReactionFunc, k8stesting.ReactionFunc) {
+		// We use a map to ensure different generation counters for different Egress
+		// resources. We do not reset the map when a resource is deleted and re-created with
+		// the same name as it is not necessary for tests (it could easily be done with
+		// delete & create reactors).
+		generation := map[string]int64{}
+		updateReactor := func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+			ua := action.(k8stesting.UpdateAction)
+			egress := ua.GetObject().(*v1alpha2.Egress)
+			generation[egress.Name] += 1
+			egress.Generation = generation[egress.Name]
+			return false, egress, nil
+		}
+		patchReactor := func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
```

Comment:
at this point, I am not so sure it is objectively better than just handling the whole change ourselves (get object from tracker, increment generation, apply patch, write object to tracker), but I don't want to overthink it too much :)